services/state/addressbook/node.proto

```diff
@@ -27,81 +34,164 @@ option java_package = "com.hederahashgraph.api.proto.java";
-     * hexadecimal characters which, translated to binary, are the SHA-384 hash of
-     * the UTF-8 NFKD encoding of the node's TLS cert in PEM format. Its value can be
-     * used to verify the node's certificate it presents during TLS negotiations.
+     * A hash of the node gRPC certificate.
+     * <p>
+     * This value MAY be used to verify the certificate presented by the node
+     * during TLS negotiation for gRPC.<br/>
+     * This value MUST be a SHA-384 hash.<br/>
+     * The TLS certificate to be hashed SHALL first be in PEM format and SHALL
+     * be encoded with UTF-8 NFKD encoding to a stream of bytes provided to
+     * the hash algorithm.<br/>
+     * This field is OPTIONAL.
      */
     bytes grpc_certificate_hash = 7;
 
     /**
-     * The consensus weight of this node in the network.
+     * A consensus weight.
+     * <p>
+     * Each node SHALL have a weight in consensus calculations.<br/>
+     * The consensus weight of a node SHALL be calculated based on the amount
+     * of HBAR staked to that node.<br/>
+     * Consensus SHALL be calculated based on agreement of greater than `2/3`
+     * of the total `weight` value of all nodes on the network.
      */
     uint64 weight = 8;
 
-    /**
-     * an enum to indicate the status of the node, not used in phase 2
+    reserved "status";
+    reserved 9;
+    /* [Placed AFTER reserved, and not doc-comment due to PBJ issue #217]
+     * Reserved as `NodeStatus status = 9;`
+     * A node status value.<br/>
+     * This value is not yet used and design is incomplete.
      */
-    //NodeStatus status = 9;
 
 }
 
+/**
+ * An enumeration of the states possible for a consensus node.
+ *
+ * Status changes are not always immediate.  Some status changes are completed
+ * in two steps, first the consensus node entry is marked for a pending
+ * change. Following the next `freeze` transaction with the field
+ * `freeze_type` set to `PREPARE_UPGRADE` (herein called a `network upgrade`)
+ * the node status change completes and the pending state is resolved.
+ *
+ * - A node is initially created in a `PENDING_ADDITION` state.
+ * - Following the next `network upgrade` every node with the status
+ *   `PENDING_ADDITION` SHALL be added to the live address book, the status
+ *   SHALL change to `IN_CONSENSUS` and these new nodes SHALL begin to operate
+ *   as full consensus nodes.
+ * - If a `deleteNode` transaction is processed for a consensus node, the node
+ *   status SHALL change to the `PENDING_DELETION` state.
+ * - Following the next `network upgrade` every node with the status
+ *   `PENDING_DELETE` SHALL be removed from the live address book, the status
+ *   SHALL change to `DELETED`, and the node SHALL no longer participate in
+ *   network consensus.
+ * - The `DELETED` state SHALL be a terminal state. Node identifiers SHALL NOT
+ *   be re-used and SHALL NOT be "revived".
+ */
```

Comment:
The entire structure makes no sense if deletion has no status.  What you're describing is a non-functional half-state, which should not be specified.
What is the intention to manage deleted nodes; that is what we need to specify and we'll adjust that _if_ future phases choose a different approach.

Reply:
Based on discussion, DELETED state is removed (the node will be removed entirely, although we'll still need to ensure node ID is never reused).
We still need the two `PENDING` status, without that we have no clear marker for what to change in the active configuration at the next upgrade.

This entity is internal network state (and part of the implementation details), so adding a status to enable managing the items modified but not yet active in network configuration is reasonable, in my opinion.

Reply:
We should remove this enum NodeStatus. It's not used in phase 2. It was designed for phase 3 to deal with Roster. These comments do not make sense. We can add them later for phase 3 if needed.

Reply:
Not sure how the comments don't make sense; they describe the behavior specified elsewhere, just specific to the status.
We aren't going to be able to implement, even phase 2, without some sort of status for pending items.
It might have been missed, but something has to track that a node is added in state but not written to configuration yet, and that something has to be _in_ state.

Reply:
In phase 2, we use a store to track those hapi transactions and they are not active until next freeze update. Here is the design [diagram](https://lucid.app/lucidchart/01f7f0ae-a6e3-4e44-9768-9220936ecec6/edit?invitationId=inv_bb15f582-8437-496f-875c-3139d5bdadfe&page=m-5o7ONTd-nK#).

Reply:
From that diagram the "store" (which is a proxy for the merkle tree) is storing `Node` objects (exactly the message that we're defining here).
Nothing in that diagram there explains how the freeze upgrade process will know _which_ `Node` objects need to be _updated_, _added_, or _deleted_.
Pending status values (and one non-pending value) seem, to me, to be the most sensible mechanism for such indicator.

Reply:
Updated based on discussion.